src/main/scala/com/exasol/cloudetl/kinesis/KinesisImportQueryGenerator.scala

```diff
@@ -1,14 +1,26 @@
 package com.exasol.cloudetl.kinesis
 
+import scala.collection.JavaConverters._
+
 import com.exasol.{ExaImportSpecification, ExaMetadata}
 import com.exasol.cloudetl.kinesis.KinesisConstants._
 
+/**
+ * This object returns an SQL query for an EXASOL IMPORT FROM SCRIPT statement.
+ */
 object KinesisImportQueryGenerator {
+
+  /** Provides a SELECT query for IMPORT.
+   *
+   * This is a function which is called inside Exasol when selecting a script based on this object.
+   */
   def generateSqlForImportSpec(
     exaMetadata: ExaMetadata,
     importSpecification: ExaImportSpecification
   ): String = {
-    val kinesisPropertiesReader = new KinesisPropertiesReader(importSpecification)
+    val kinesisPropertiesReader = new KinesisPropertiesReader(
+      importSpecification.getParameters.asScala.toMap
+    )
```

Comment:
Use the `properties.mkString()` to generate the key-value string and provide it to the udfs.

Reply:
done